tool/internal_pkg/generator/type.go

```diff
@@ -177,17 +197,41 @@ func (t *Task) Build() error {
 	if err != nil {
 		return err
 	}
-	for _, n := range templateNames {
-		x, err = x.Parse(templateExtensions[n])
-		if err != nil {
-			return fmt.Errorf("failed to parse extension %q for %s: %w (%#q)",
-				n, t.Name, err, templateExtensions[n])
+	// old fashion
+	x, err = applyExtension(t.Name, x)
+	if err != nil {
+		return err
+	}
+	// new fashion
+	for _, str := range t.makeExtension() {
+		if x, err = x.Parse(str); err != nil {
+			return err
 		}
 	}
+
 	t.Template = x
 	return nil
 }
 
+var indexes = map[string]int{
+	ClientFileName: 0, ServerFileName: 2, InvokerFileName: 4,
+}
```

Comment:
这里硬编码是不是不太好，改成字符串判断是不是好一些
```suggestion
func fileTemplateExtension(name string) (option, eof string) {
	for _, tn := range templateNames {
		if strings.HasPrefix(tn, "@"+name+"-") {
			if strings.HasSuffix(tn, "-option") {
				option = tn
			} else if strings.HasSuffix(tn, "-eof") {
				eof = tn
			}
		}
	}
	return
}
```